### DISCUSSION TITLE
Auto-importing function* exports from /utils

### DISCUSSION THREAD

**Discussion by AloisSeckar** [nuxt/nuxt#20565] [Ideas]
Playing around with my series of Nuxt examples I have just found out, that unlike regular functions, definitions of [JS Generators](https://javascript.info/generators) don't get auto-imported when declared inside `/utils` directory.

- definition [here](https://github.com/AloisSeckar/demos-nuxt/blob/main/nuxt-utils/utils/utils.ts)
- usage [here](https://github.com/AloisSeckar/demos-nuxt/blob/main/nuxt-utils/pages/index.vue)

Without explicit import `generateSequence is not defined` error occurs.

Is there a technical reason for this? Or is it just something not considered important so far? (because tbh, if I didn't attend a talk about generators on JS conference recently, I didn't even know they exist :D)

**Comment by warflash** [comment]
Sounds like you may want to raise an issue to figure out whether there's a reason for it (that would need to be added to the docs) or whether it's a bug

**Reply by AloisSeckar** [reply]
First I wanted to clarify if there isn't any known and "obvious" reason why not to do it. This is why I started a discussion first.
